doc/api/perf_hooks.md

```diff
@@ -0,0 +1,559 @@
+* `perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR` (2)
+* `perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL` (4)
+* `perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB` (8)
+
+### performanceGC.name
+<!-- YAML
+added: REPLACEME
+-->
+* {string}
+
+The name of the `PerformanceEntry`. For `PerformanceGC` instances, this will
+always be `'gc'`.
+
+### performanceGC.startTime
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp marking the starting time of the
+garbage collection operation.
+
+## Class: PerformanceNodeTiming
+<!-- YAML
+added: REPLACEME
+-->
+
+A subclass of `PerformanceEntry` that provides timing metrics specific to the
+Node.js runtime.
+
+### performanceNodeTiming.arguments
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp at which the Node.js command-line
+arguments processing completed.
+
+### performanceNodeTiming.bootstrapComplete
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp at which the Node.js process
+completed bootstrap.
+
+### performanceNodeTiming.duration
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The total number of milliseconds that the Node.js process has been running.
+
+### performanceNodeTiming.initialize
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp at which the Node.js platform was
+initialized.
+
+### performanceNodeTiming.inspectorStart
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp at which the Node.js inspector
+agent completed startup.
+
+### performanceNodeTiming.loopExit
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp at which the Node.js event loop
+exited.
+
+### performanceNodeTiming.loopStart
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The high resolution millisecond timestamp at which the Node.js event loop
+started.
+
+## Class: PerformanceObserver(callback)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `callback` {Function} A `PerformanceObserverCallback` callback function.
+
+`PerformanceObserver` objects provide notifications when new
+`PerformanceEntry` instances have been added to the Performance Timeline.
+
+```js
+const {
+  performance,
+  PerformanceObserver
+} = require('perf_hooks');
+
+const obs = new PerformanceObserver((list, observer) => {
+  console.log(list.getEntries());
+  observer.disconnect();
+});
+obs.observe({ entryTypes: ['mark'], buffered: true });
+
+performance.mark('test');
+```
+
+Because `PerformanceObserver` instances introduce their own additional
+performance overhead, instances should not be left subscribed to notifications
+indefinitely. Users should disconnect observers as soon as they are no
+longer needed.
+
+### Callback: PerformanceObserverCallback(list, observer)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `list` {PerformanceEntryList}
```

Comment:
This should be named `PerformanceObserverEntryList` per [spec](https://w3c.github.io/performance-timeline/#dom-performanceobservercallback). [`PerformanceEntryList`](https://w3c.github.io/performance-timeline/#dom-performanceentrylist) is just a typedef for a JS array